rtt_roscomm/include/rtt_roscomm/passthrough_callback_queue.hpp

```diff
@@ -0,0 +1,232 @@
+
+#ifndef RTT_ROSCOMM__PASSTHROUGH_CALLBACK_QUEUE_HPP_
+#define RTT_ROSCOMM__PASSTHROUGH_CALLBACK_QUEUE_HPP_
+
+#include <ros/ros.h>
+#include <ros/callback_queue_interface.h>
+#include <ros/subscription_queue.h>
+#include <ros/callback_queue.h>
+
+namespace rtt_roscomm {
+  class PassthroughCallbackQueue: public ros::CallbackQueueInterface
+  {
+    public:
+      PassthroughCallbackQueue();
+
+      /** 
+       * Implementation of ros::CallbackQueueInterface::addCallback()
+       * 
+       * @param callback aaa
+       * @param owner_id aaa
+       * 
+       * @return void
+       */
+      virtual void addCallback(
+          const ros::CallbackInterfacePtr &callback,
+          uint64_t owner_id=0);
+
+      /** 
+       * Implementation of ros::CallbackQueueInterface::removeByID()
+       * 
+       * @param owner_id aaa
+       * 
+       * @return void
+       */
+      virtual void removeByID(uint64_t owner_id);
+
+    private:
+
+      /**
+       * Port where to write the message
+       */
+      // base::PortInterface* output_port_ptr_;
+
+  }; // class PassthroughCallbackQueue
+
+  /**
+   * Helpers to generate subscribers with a custom ros::CallbackQueueInterface
+   */
```

Comment:
Not needed. This special implementation of `ros::CallbackQueueInterface` is used in exactly one place, in [`RosSubChannelElement::RosSubChannelElement()`](https://github.com/orocos/rtt_ros_integration/pull/156/files#diff-b7dd3168b5e27cadacbc7d8c54c72e8dR257) below. It should not be used anywhere else, because in most other contexts (with arbitrary user-defined callbacks) it won't do anything good.

The implementation of the constructor is already kind of unreadable, and using a `ros::SubscribeOptions ops` instance there would improve that anyway (see next comment). So all those helper functions should be removed. They are also not specific to `PassthroughCallbackQueue`, and last but not least would need to be maintained for future updates in the roscpp API that add or modify overloads.

Please see the suggested implementation of `RosSubChannelElement::RosSubChannelElement()` below.